test/unit/Container.spec.ts

```diff
@@ -98,4 +113,14 @@ describe("Container", function() {
         Container.get(ExtraService).byeMessage.should.be.equal("buy world");
     });
 
+    it("should support class factories", function() {
```

Comment:
Yes, that makes sense. What if we introduce a namespaces concept or a named containers then?

```ts
// This is the container without a name, it's a default one:
const container1 = Container.create();

// This is additional container which goes under the "second-container" name:
const container2 = Container.create("second-container");

// This service belongs to the default container.
@Service()
class Foo {
}

// However, this service belongs to the second container.
@Service("bar", "second-container")
class Bar {
}

```

Or more meaningful API:

```ts
@Service({
  name: "bar",
  container: "second-container"
})
class Bar {
}
```

In order to replace a container:

```ts
const container1 = Container.create();
const container2 = Container.create("second-container");

// Use one of these interchangeably:
Container.replace("default", "second-container");
Container.replace("default", container2);
Container.replace(container1, container2);
// "second-container" is now used as a default, the default one is effectively destroyed

// These two will replace the default container with a new empty one.
const newDefault = Container.replace("default");
const newDefault = Container.replace();
```

This looks like a viable solution, but it also looks a bit complicated. What do you think?